### PR TITLE
feat(node): new Full Node type

### PR DESCRIPTION
**Pull request by Wondertan**
Just a new Full Node type with Full availability

### CODE REVIEW

File: node/components.go [hunk 1]
```diff
@@ -26,6 +27,18 @@ func bridgeComponents(cfg *Config, store Store) fxutil.Option {
 		fxutil.Supply(Bridge),
 		baseComponents(cfg, store),
 		nodecore.Components(cfg.Core, store.Core),
+		fxutil.Provide(services.LightAvailability), // TODO(@Wondertan): Remove strict requirements to have Availability
```
Comment (renaynay) [inline node/components.go:30]: This requires us to change `ShareService` not to take `Availability` right?

File: node/components.go [hunk 2]
```diff
@@ -26,6 +27,18 @@ func bridgeComponents(cfg *Config, store Store) fxutil.Option {
...
+	)
+}
+
+// bridgeComponents keeps all the components as DI options required to build a Full Node.
```
Comment (renaynay) [inline node/components.go:34]:
```suggestion
// fullComponents keeps all the components as DI options required to build a Full Node.
```

File: node/type.go [hunk 3]
```diff
@@ -8,9 +8,11 @@ const (
 	// Bridge is a Celestia Node that bridges the Celestia consensus network and data availability network.
 	// It maintains a trusted channel/connection to a Celestia Core node via the core.Client API.
 	Bridge Type = iota + 1
-	// Light is a stripped-down Celestia Node which aims to be lightweight while preserving highest possible
+	// Light is a stripped-down Celestia Node which aims to be lightweight while preserving the highest possible
 	// security guarantees.
 	Light
+	// Full is a Celestia Node which keeps track of full-blown blocks.
```
Comment (renaynay) [inline node/type.go:14]:
```suggestion
	// Full is a Celestia Node that stores blocks in their entirety.
```

**Review by renaynay — COMMENTED** [review]
tiny nits, LGTM thankie thankie

File: node/components.go (hunk 1 above)
Comment (Wondertan) [reply node/components.go:30]: Yes

**Comment by Wondertan** [comment]
No merge until next release.

**Review by liamsi — APPROVED** [review]
Straightforward. Nothing to complain about :)

**Review by renaynay — APPROVED** [review]

**Pull request merged**